Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Controller.mo

```diff
@@ -0,0 +1,2498 @@
+    final max=fill(1, nChi),
+    final unit=fill("1", nChi))
+    "Head pressure control valve position"
+    annotation (Placement(transformation(extent={{920,220},{960,260}}),
+      iconTransformation(extent={{100,70},{140,110}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yConWatPumSpe[nConWatPum](
+    final unit=fill("1", nChi),
+    final min=fill(0, nChi),
+    final max=fill(1, nChi))
+    "Condenser water pump speed setpoint"
+    annotation (Placement(transformation(extent={{920,130},{960,170}}),
+        iconTransformation(extent={{100,40},{140,80}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yChiWatMinFloSet(
+    final quantity="VolumeFlowRate",
+    final unit="m3/s",
+    final min=0) "Chilled water minimum flow setpoint"
+    annotation (Placement(transformation(extent={{920,100},{960,140}}),
+      iconTransformation(extent={{100,12},{140,52}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput yConWatPum[nConWatPum]
+    "Status setpoint of condenser water pump"
+    annotation (Placement(transformation(extent={{920,70},{960,110}}),
+      iconTransformation(extent={{100,-20},{140,20}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yChiWatIsoVal[nChi](
+    final unit=fill("1", nChi),
+    final min=fill(0, nChi),
+    final max=fill(1, nChi))
+    "Chiller isolation valve position setpoints"
+    annotation (Placement(transformation(extent={{920,-20},{960,20}}),
+        iconTransformation(extent={{100,-90},{140,-50}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput yReaChiDemLim
+    if need_reduceChillerDemand
+    "Release chiller demand limit, normally true"
+    annotation (Placement(transformation(extent={{920,-230},{960,-190}}),
+        iconTransformation(extent={{100,-60},{140,-20}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yMinValPosSet(
+    final min=0,
+    final max=1,
+    final unit="1") "Chilled water minimum flow bypass valve position setpoint"
+    annotation (Placement(transformation(extent={{920,-340},{960,-300}}),
+        iconTransformation(extent={{100,-120},{140,-80}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yTowCelIsoVal[nTowCel](
+    final min=fill(0, nTowCel),
+    final max=fill(1, nTowCel),
+    final unit=fill("1", nTowCel))
+    "Cooling tower cells isolation valve position"
+    annotation (Placement(transformation(extent={{920,-640},{960,-600}}),
+      iconTransformation(extent={{100,-200},{140,-160}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput yTowCel[nTowCel]
+    "Vector of tower cells status setpoint"
+    annotation(Placement(transformation(extent={{920,-680},{960,-640}}),
+      iconTransformation(extent={{100,-230},{140,-190}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yTowFanSpe[nTowCel](
+    final min=fill(0, nTowCel),
+    final max=fill(1, nTowCel),
+    final unit=fill("1", nTowCel))
+    "Fan speed setpoint of each cooling tower cell"
```

Comment:
- Should be a scalar (unique control point for all CT)
- Should be conditional to water-cooled plant